wagtailsharing/models.py

```diff
@@ -44,3 +45,17 @@ def root_url(self):
             return "https://{}".format(self.hostname)
         else:
             return "http://{}:{:d}".format(self.hostname, self.port)
+
+
+class ShareableRoutablePageMixin(RoutablePageMixin):
+    def route(self, request, path_components):
+        if getattr(request, "routed_by_wagtail_sharing", False):
+            page = self.get_latest_revision_as_page()
+            return super(ShareableRoutablePageMixin, page).route(
+                request, path_components
+            )
+        return super().route(request, path_components)
+
+
+class ShareableRoutablePage(ShareableRoutablePageMixin, Page):
```

Comment:
Can this be marked with [`abstract = True`](https://github.com/wagtail/wagtail/blob/d18ffb0233dfb01c5fc876fa7e9b755aa9a256f0/wagtail/contrib/routable_page/models.py#L180) like `RoutablePage` to prevent it from creating a table in the database unless used by a concrete child model? I realize this will require creating a separate model for the tests, but it feels worth it to avoid a new table unnecessarily.

Similarly, should this be mentioned in the docs above?

Reply:
I'll add a test app with a test model.

Reply:
Test model added in 32494ac.